mbuild/utils/conversion.py

```diff
@@ -78,16 +92,46 @@ def OPLS_to_RB(f1, f2, f3, f4):
 
     Parameters
     ----------
-    f1, f2, f3, f4 : OPLS dihedrals coeffs (in kcal/mol)
+    f0, f1, f2, f3, f4 : OPLS dihedrals coeffs (in kcal/mol)
+    error_tol : float, default=1e-4
+        The acceptable absolute tolerance, which the f0 term can be from zero
+        without throwing a warning.
 
     Returns
     -------
     RB_coeffs : np.array, shape=(6,)
         Array containing the Ryckaert-Bellemans dihedrals
         coeffs c0, c1, c2, c3, c4, and c5 (in kcal/mol)
 
+    Notes
+    -----
+    NOTE: fO IS TYPICALLY NOT IN THE OPLS DIHEDRAL EQUATION (i.e., f0=0).
+
+    WARNING: The f0 term is the constant for the OPLS dihedral equation,
+    which is converted and added to a constant for the RB torsions equation via the c0 coefficient.
+    If the f0 term is zero in the OPLS dihedral form or set to zero in this equation,
+    the dihedral is not an exact conversion;
+    since this constant does not contribute to the force equation,
+    this should provide matching results for MD, but the energy for each
+    dihedral will be shifted by the real f0/2 value.
     """
-    c0 = f2 + (f1 + f3) / 2
+    try:
+        error_tol = abs(float(error_tol))
+    except:
+        raise TypeError("ERROR: The error_tol variable must be a float.")
+
+    if np.all(np.isclose(f0, 0, atol=error_tol, rtol=0)):
```

Comment:
I agree with @CalCraven that these f0 != 0.0 should initially raise an error, and the user can choose to turn that off to a warning by changing a flag. But i think the default behavior should be to error out.

Reply:
Added this option calling it value_error_out_of_tol=True as default.